src/traceroute/android.cpp

```diff
@@ -113,6 +119,8 @@ void AndroidProber::send_probe(std::string addr, int port, int ttl,
   socklen_t sslen;
   timeval tv;
 
+  init();
```

Comment:
This is not correct: you don't want to `init()` every time you fire off a probe. What you want is to emulate the constructor semantic, but opportunistically; i.e., you want to run `init()` only once when it's needed to do so. An alternative -- and perhaps more robust approach -- is that you make `init()` idempotent, meaning that it only initialize stuff if it's needed otherwise it just returns immediately.